File: c/cz/mg/vulkan/PFN_vkSubmitDebugUtilsMessageEXT.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkSubmitDebugUtilsMessageEXT_call(JNIEnv* env, jclass jc, jlong address, jlong instance, jlong messageSeverity, jlong messageTypes, jlong pCallbackData){
    (void)env;
    (void)jc;
    PFN_vkSubmitDebugUtilsMessageEXT f = (PFN_vkSubmitDebugUtilsMessageEXT)jniLongToFunctionPointer(address);
    f(
        *((VkInstance*)jniLongToPointer(instance)),
        *((VkDebugUtilsMessageSeverityFlagBitsEXT*)jniLongToPointer(messageSeverity)),
        *((VkDebugUtilsMessageTypeFlagsEXT*)jniLongToPointer(messageTypes)),
        ((VkDebugUtilsMessengerCallbackDataEXT*)jniLongToPointer(pCallbackData))
    );
}

File: cz/mg/vulkan/PFNvkSubmitDebugUtilsMessageEXT.java
package cz.mg.vulkan;

public class PFNvkSubmitDebugUtilsMessageEXT extends VkFunctionPointer {
    public PFNvkSubmitDebugUtilsMessageEXT() {
    }

    public PFNvkSubmitDebugUtilsMessageEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkSubmitDebugUtilsMessageEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkSubmitDebugUtilsMessageEXT(long value) {
        setValue(value);
    }

    public PFNvkSubmitDebugUtilsMessageEXT(VkInstance instance) {
        super(instance, new VkString("vkSubmitDebugUtilsMessageEXT"));
    }

    public void call(VkInstance instance, VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity, VkDebugUtilsMessageTypeFlagsEXT messageTypes, VkDebugUtilsMessengerCallbackDataEXT pCallbackData){
        call(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getSinkAddress(), messageSeverity != null ? messageSeverity.getVkAddress() : VkPointer.getSinkAddress(), messageTypes != null ? messageTypes.getVkAddress() : VkPointer.getSinkAddress(), pCallbackData != null ? pCallbackData.getVkAddress() : VkPointer.NULL);
    }

    private static native void call(long vkaddress, long instance, long messageSeverity, long messageTypes, long pCallbackData);
}
